browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/UserInfoActivity.java

```diff
@@ -85,114 +102,169 @@ protected void onCreate(Bundle savedInstanceState) {
+                                    mSessionClient.clear();
+                                    finish();
+                                }
+                                continuePresentationData();
+                            }
+                        });
+                        break;
+                    case NOT_BLOCKED:
+                        showMessage(getString(R.string.setup_lockscreen_info_msg));
+                        break;
+                    case NOT_SUPPORTED:
+                        showMessage(getString(R.string.hardware_not_support_fingerprint));
+                        break;
+                    case NO_FINGERPRINTS:
+                        showMessage(getString(R.string.fingerprint_not_enrolled));
+                        break;
+                }
+            } catch (Exception e) {
+                showMessage(e.getMessage());
+            }
+        });
+        alertBuilder.setNegativeButton(R.string.no, (dialog, which) -> {
+            continuePresentationData();
+            dialog.dismiss();
+        });
+        alertBuilder.show();
     }
 
     private void fetchUserInfo() {
         oktaProgressDialog.show(getString(R.string.user_info_loading));
-        mOktaAppAuth.getUserInfo(new OktaAppAuth.OktaAuthActionCallback<JSONObject>() {
-            @Override
-            public void onSuccess(JSONObject jsonObject) {
-                runOnUiThread(() -> {
-                    oktaProgressDialog.hide();
-                    mUserInfoJson.set(jsonObject);
-                    displayAuthorizationInfo();
-                });
-            }
 
+        mSessionClient.getUserProfile(new RequestCallback<UserInfo, AuthorizationException>() {
             @Override
-            public void onTokenFailure(@NonNull AuthorizationException e) {
-                runOnUiThread(() -> {
-                    oktaProgressDialog.hide();
-                    mUserInfoJson.set(null);
-                    displayAuthorizationInfo();
-                    showMessage("TokenFailure: "+e.errorDescription);
-                });
+            public void onSuccess(UserInfo userInfo) {
+                oktaProgressDialog.hide();
+                mUserInfoJson.set(userInfo);
+                displayAuthorizationInfo();
             }
 
             @Override
-            public void onFailure(int i, Exception e) {
-                runOnUiThread(() -> {
-                    oktaProgressDialog.hide();
-                    mUserInfoJson.set(null);
-                    displayAuthorizationInfo();
-                    showMessage("Failure: "+e.getMessage());
-                });
+            public void onError(String s, AuthorizationException e) {
+                oktaProgressDialog.hide();
+                mUserInfoJson.set(null);
+                showMessage("Failure fetch user profile: " + e.error + ":" + e.errorDescription);
+                if (e.type == AuthorizationException.TYPE_PERSISTENCE_ERROR) {
```

Comment:
We have to change what the SDK returns in this case. we should use a TYPE explains it is a `UserNotAuthenticatedException` have AuthorizationException include that. `TYPE_PERSISTENCE_ERROR` doesn't fit with the actual error which is a decryption error due to the user not being authenticated. This has nothing to do with persisting data.